myst_nb/parser.py

```diff
@@ -13,45 +13,56 @@
 from markdown_it.rules_core import StateCore
 from markdown_it.utils import AttrDict
 
-from myst_parser.main import default_parser
+from myst_parser.main import default_parser, to_docutils
 from myst_parser.sphinx_renderer import SphinxRenderer
 from myst_parser.sphinx_parser import MystParser
 
+from myst_nb.cache import add_notebook_outputs
+from myst_nb.converter import string_to_notebook
 from myst_nb.nb_glue import GLUE_PREFIX
 from myst_nb.nb_glue.domain import NbGlueDomain
 
-from myst_nb.cache import add_notebook_outputs
-
 
 SPHINX_LOGGER = logging.getLogger(__name__)
 
 
 class NotebookParser(MystParser):
     """Docutils parser for IPynb + CommonMark + Math + Tables + RST Extensions """
 
-    supported = ("ipynb",)
+    supported = ("myst-nb",)
     translate_section_name = None
 
     default_config = {"known_url_schemes": None}
 
-    config_section = "ipynb parser"
+    config_section = "myst-nb parser"
     config_section_dependencies = ("parsers",)
 
-    def parse(self, inputstring: str, document: nodes.document, source_path=None):
+    def parse(self, inputstring: str, document: nodes.document):
 
-        # de-serialize the notebook
-        ntbk = nbf.reads(inputstring, nbf.NO_CONVERT)
         self.reporter = document.reporter
+        self.env = document.settings.env
         self.config = self.default_config.copy()
-
         try:
             new_cfg = document.settings.env.config.myst_config
             self.config.update(new_cfg)
         except AttributeError:
             pass
+
+        try:
+            ntbk = string_to_notebook(inputstring, self.env)
+        except Exception as err:
+            SPHINX_LOGGER.error(
+                "Notebook load failed for %s: %s", self.env.docname, err
+            )
+            return
+        if not ntbk:
+            # Read the notebook as a text-document
+            to_docutils(inputstring, options=self.config, document=document)
+            return
+
         # add outputs to notebook from the cache
         if document.settings.env.config["jupyter_execute_notebooks"] != "off":
```

Comment:
```suggestion
        if self.env.config["jupyter_execute_notebooks"] != "off":
```

it seemed you were doing this elsewhere

Reply:
any reason not to commit this? I'll merge once it's resolved one way or another